src/components/card/card.jsx

```diff
@@ -1,22 +1,35 @@
 import React, {PureComponent} from "react";
 import PropTypes from "prop-types";
 import {capitalize, formatRating} from "../../utils/common.js";
-import {PlaceType} from "../../consts.js";
+import {PlaceType, CardType} from "../../consts.js";
 
 class Card extends PureComponent {
 
   render() {
-    const {card, onMouseEnter, onTitleClick} = this.props;
+    const {card, cardType, onMouseEnter, onTitleClick} = this.props;
     const placeTypeName = capitalize(card.type);
+    let cardClassName = ``;
+    let imageWrapperClassName = ``;
+
+    switch (cardType) {
+      case CardType.CITY:
+        cardClassName = `cities__place-card`;
+        imageWrapperClassName = `cities__image-wrapper`;
+        break;
+      case CardType.NEAR:
+        cardClassName = `near-places__card`;
+        imageWrapperClassName = `near-places__image-wrapper`;
+        break;
+    }
```

Comment:
Слишком громоздкая конструкция получилась. Я правильно понимаю, что может быть только 2 типа карточки? Если да, то юзай тернарный оператор.

```suggestion
    const cardClassName = cardType === CardType.CITY ? `cities__place-card` : `near-places__card`;
    const imageWrapperClassName = cardType === CardType.CITY ? `cities__image-wrapper` : `near-places__image-wrapper`;
```

Reply:
Там есть еще список избранного, предполагаю, что его таким же образом буду создавать

Reply:
Тогда лучше сделать enum. Тип карточки: класс

Reply:
Я правильно понял, что ты предлагаешь сделать два enum - для классов карточки и для классов обертки картинки? Если нет, можешь показать пример что ты имел в виду?

Reply:
Да, ты понял все правильно. Это позволит исключить свитч/кейс.

Reply:
Чет я мальца затупил, но не пойму, как использование enum позволит тут отказаться от свитч/кейс тут

Reply:
const cardClassNames = {
CardType.CITY: 'cities__place-card',
CardType.NEAR: 'near-places__card',
};

и использовать вот так: cardClassNames[cardType]

Reply:
А конструкции вида CardType.CITY: внутри enum не противоречат критерию Б8? (https://up.htmlacademy.ru/react/4/criteries#b8)
У меня создалось впечатление, что имена полей (или как это правильно назвать) должны быть заглавными буквами
О том, чтобы писать там (грубо говоря) 'city' вместо CITY я тоже подумал уже, но не понял, можно ли так по критериям делать

Reply:
Ну если скопировать полностью мой пример – то да, противоречит. Надо дотюнить немного)

Reply:
Ну если скопировать полностью твой пример, там не только линтер будет ругаться, там и консоль ошибку выдает) Я чет вообще не вкуриваю, как предполагается дотюнить этот пример, чтобы и enum, и критерий соблюден, и без свитча

Могу ли я вообще делать так:
```
const CardClass = {
  'city': 'cities__place-card',
  'near': 'near-places-card',
}
```
потому что это выглядит самым логичным решением

Reply:
```
export const CardTypeClassNames = {
  [CardType.CITY]: `cities__place-card`,
  [CardType.NEAR]: `near-places-card`,
};
```